powerstore/volume_helper.go

```diff
@@ -2,15 +2,28 @@ package powerstore
 
 import (
 	"context"
+	"fmt"
 	pstore "github.com/dell/gopowerstore"
+	"terraform-provider-powerstore/client"
 	"terraform-provider-powerstore/models"
 )
 
+const (
+	// MiB to convert size in megabytes
+	MiB = 1024 * 1024
+	// GiB to convert size in gigabytes
+	GiB = 1024 * MiB
+	// TiB to convert size in terabytes
+	TiB = 1024 * GiB
+)
+
 func updateVolState(volState *models.Volume, volResponse pstore.Volume, hostMapping []pstore.HostVolumeMapping, volPlan *models.Volume, operation string) {
 	// Update value from Volume Response to State
 	volState.ID.Value = volResponse.ID
 	volState.Name.Value = volResponse.Name
-	volState.Size.Value = volResponse.Size
+	size, unit := convertFromBytes(volResponse.Size)
+	volState.Size.Value = size
+	volState.CapacityUnit.Value = unit
```

Comment:
For line 30, `if operation == "Create" || operation == "Update" `
Better way would be plan Modifiers,  whether independent for every attribute or ResourceWithPlanModifier for everyone at single place